Source/WTF/wtf/haiku/RunLoopHaiku.cpp

```diff
@@ -115,7 +115,8 @@ void RunLoop::stop()
 
 void RunLoop::wakeUp()
 {
-    m_handler->Looper()->PostMessage('loop', m_handler);
+	RunLoop::current().performWork();
```

Comment:
Uh, this look completely wrong!
RunLoop::wakeUp is called from other threads to wake up the loop.
performWork should be run inside the main loop thread.

There is nothing to win in taking such shortcuts. You will only get things to crash in very strange ways.

Also, this probably breaks existing use of runloop in WebKitLegacy. It would be great if I could merge your changes in the main branch, but with changes like this, it's not going to work.

Reply:
Sorry just was curious what happens beyond this point 😢 but i knew it is going to break things i will find howto make it work.